File: python/meteroid/models/customer_details.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime

from ..serialization import BaseModel
from .address import Address
from .customer_id import CustomerId


@dataclasses.dataclass
class CustomerDetails(BaseModel):
    id: CustomerId

    name: str

    snapshot_at: datetime

    alias: t.Optional[str] = None

    billing_address: t.Optional[Address] = None

    email: t.Optional[str] = None

    vat_number: t.Optional[str] = None

File: go/customer_details.go
// this file is @generated
package meteroid

import "time"

type CustomerDetails struct {
	Alias *string `json:"alias,omitempty"`

	BillingAddress *Address `json:"billing_address,omitempty"`

	Email *string `json:"email,omitempty"`

	Id CustomerId `json:"id"`

	Name string `json:"name"`

	// RFC 3339 timestamp.
	SnapshotAt time.Time `json:"snapshot_at"`

	VatNumber *string `json:"vat_number,omitempty"`
}
